src/parser/validate.ts

```diff
@@ -99,6 +71,96 @@ export function validate(parse: {
     }
   }
 
+  async function resolveFlags(flags: FlagRelationship[]): Promise<Record<string, unknown>> {
+    const promises = flags.map(async flag => {
+      if (typeof flag === 'string') {
+        return [flag, parse.output.flags[flag]]
+      }
+
+      const result = await flag.when(parse.output.flags)
+      return result ? [flag.name, parse.output.flags[flag.name]] : null
+    })
+    const resolved = await Promise.all(promises)
+    return Object.fromEntries(resolved.filter(r => r !== null) as [string, unknown][])
+  }
+
+  async function validateExclusive(name: string, flags: FlagRelationship[]) {
+    const resolved = await resolveFlags(flags)
+    const keys = Object.keys(resolved).reduce((acc, key) => {
+      if (resolved[key]) acc.push(key)
+      return acc
+    }, [] as string[])
```

Comment:
not sure why this reduce needs to be here.  Another design would be

Object.entries(resolved).forEach(

and use 1 && conditional to check what you're checking to get to an error
1. the value is truthy AND
2.`parse.output.metadata.flags[flag]?.setFromDefault` is falsy AND
3. `parse.output.metadata.flags[name]?.setFromDefault` is falsy AND
4. `parse.output.flags[flag]` is truthy
--> throw!

that way it's only 1 iteration, the conditionals can be in whatever order is most likely to resolve soonest and it'll fail on the first error